basm.go

```diff
@@ -0,0 +1,60 @@
+package basm
+
+// WriteToHost takes a byte slice and writes it to the shared memory of the host's
+// WebAssembly runtime. It returns a packed offset/size pair in a format
+// compatible with WebAssembly numeric types. The host is expected to free the
+// memory when it is done with it.
+func WriteToHost(data []byte) uint64 {
+	return leakToSharedMem(data)
+}
+
+// ReadFromHost takes a packed offset/size pair in a format compatible with
+// WebAssembly numeric types and returns the byte slice that was written to the
+// shared memory of the host's WebAssembly runtime. The host is expected to free
+// the memory when the guest function is complete.
+func ReadFromHost(inputPtr uint64) []byte {
+	return bytesFromFatPtr(inputPtr)
+}
+
+// Log writes a message to the logs returned to the user with each invocation.
+func Log(msg string) {
+	hostFuncBufferLog(msg)
+}
+
+// HTTPRequest uses the host's HTTP client to make a request to the given URL.
+func HTTPRequest(req HTTPRequestInput) (HTTPRequestOutput, error) {
+	resp, err := hostFuncHTTPRequest(httpRequestInput(req))
+	return HTTPRequestOutput(resp), err
```

Comment:
The fact that the exported function argument/return types are castable to the unexported internal types is coincidental and may diverge. LMK if you think this should get pushed into the unexported sdk functions.

Reply:
I moved this behavior to a descriptively named helper function so it should be obvious what is going on. 

The json tags were a leftover from when the exported and unexported input/output types were the same struct. Removed